File: day1/day1.rs
use std::fs::File;
use std::io::{self, prelude::*, BufReader};

fn main() -> io::Result<()> {
    let file = File::open("./input.txt")?;
    let reader = BufReader::new(file);
    let packs: &mut Vec<i32> = &mut Vec::new();
    let mut curr = 0;

    for line in reader.lines() {
      let row = line?;
      if row.eq("") {
          packs.push(curr);
          curr = 0;
          continue;
        }
        let val = row.parse::<i32>().unwrap();
      curr += val;
    }

    let s = packs.len();
    packs.sort();

    println!("{}", packs[s-1] + packs[s-2] + packs[s-3]);
    Ok(())
}

File: day1/day1.kt
import java.io.File

fun main() {
    var lines = File("input.txt").readLines()
    var subSum = 0
    var packins = ArrayList<Int>()

    for (line in lines) {
        if (line.equals("")) {
            packins.add(subSum)
            subSum = 0
        } else {
            subSum += Integer.parseInt(line)
        }
    }

    packins.sort()
    packins.reverse()

    println(packins[0])
    println(packins[0] + packins[1] + packins[2])
}
